### ISSUE TITLE
Investigate issue with pyparsing 2.1.6

### ISSUE THREAD

**Issue by jenshnielsen** [matplotlib/matplotlib#6932]
Travis is failing with pyparsing 2.1.6 workaround in #6931

```
======================================================================
ERROR: matplotlib.tests.test_mathtext.test_mathtext_exceptions
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/matplotlib/matplotlib/lib/matplotlib/tests/test_mathtext.py", line 224, in test_mathtext_exceptions
    parser.parse(math)
  File "/home/travis/build/matplotlib/matplotlib/lib/matplotlib/mathtext.py", line 3285, in parse
    box = self._parser.parse(s, font_output, fontsize, dpi)
  File "/home/travis/build/matplotlib/matplotlib/lib/matplotlib/mathtext.py", line 2513, in parse
    result = self._expression.parseString(s)
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1594, in parseString
    loc, tokens = self._parse( instring, 0 )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1501, in _parseCache
    value = self._parseNoCache(instring, loc, doActions, callPreParse)
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1366, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 3567, in parseImpl
    return self.expr._parse( instring, loc, doActions, callPreParse=False )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1501, in _parseCache
    value = self._parseNoCache(instring, loc, doActions, callPreParse)
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1366, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 3238, in parseImpl
    loc, exprtokens = e._parse( instring, loc, doActions )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1501, in _parseCache
    value = self._parseNoCache(instring, loc, doActions, callPreParse)
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1366, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 3772, in parseImpl
    return super(ZeroOrMore, self).parseImpl(instring, loc, doActions)
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 3722, in parseImpl
    loc, tokens = self_expr_parse( instring, loc, doActions, callPreParse=False )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1501, in _parseCache
    value = self._parseNoCache(instring, loc, doActions, callPreParse)
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1366, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 3221, in parseImpl
    loc, resultlist = self.exprs[0]._parse( instring, loc, doActions, callPreParse=False )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1501, in _parseCache
    value = self._parseNoCache(instring, loc, doActions, callPreParse)
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1392, in _parseNoCache
    tokens = fn( instring, tokensStart, retTokens )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1036, in wrapper
    ret = func(*args[limit[0]:])
  File "/home/travis/build/matplotlib/matplotlib/lib/matplotlib/mathtext.py", line 2584, in math_string
    return self._math_expression.parseString(toks[0][1:-1])
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1594, in parseString
    loc, tokens = self._parse( instring, 0 )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1501, in _parseCache
    value = self._parseNoCache(instring, loc, doActions, callPreParse)
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1366, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 3567, in parseImpl
    return self.expr._parse( instring, loc, doActions, callPreParse=False )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1501, in _parseCache
    value = self._parseNoCache(instring, loc, doActions, callPreParse)
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1366, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 3722, in parseImpl
    loc, tokens = self_expr_parse( instring, loc, doActions, callPreParse=False )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1501, in _parseCache
    value = self._parseNoCache(instring, loc, doActions, callPreParse)
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1366, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 3567, in parseImpl
    return self.expr._parse( instring, loc, doActions, callPreParse=False )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1501, in _parseCache
    value = self._parseNoCache(instring, loc, doActions, callPreParse)
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1366, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 3373, in parseImpl
    ret = e._parse( instring, loc, doActions )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1501, in _parseCache
    value = self._parseNoCache(instring, loc, doActions, callPreParse)
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1366, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 3567, in parseImpl
    return self.expr._parse( instring, loc, doActions, callPreParse=False )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1501, in _parseCache
    value = self._parseNoCache(instring, loc, doActions, callPreParse)
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1366, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 3373, in parseImpl
    ret = e._parse( instring, loc, doActions )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1501, in _parseCache
    value = self._parseNoCache(instring, loc, doActions, callPreParse)
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1366, in _parseNoCache
    loc,tokens = self.parseImpl( instring, preloc, doActions )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 3567, in parseImpl
    return self.expr._parse( instring, loc, doActions, callPreParse=False )
  File "/home/travis/build/matplotlib/matplotlib/venv/lib/python2.7/site-packages/pyparsing.py", line 1504, in _parseCache
    cache.set(lookup, pe.__class__(*pe.args))
TypeError: __init__() takes exactly 2 arguments (4 given)
----------------------------------------------------------------------
```

Need to figure out if this is a bug that needs reporting upstream or if we are doing anything wrong

**Comment by jenshnielsen** [comment]
Also given that this is the third time within a relatively short time span that a new version of pyparsing breaks we should perhaps consider limiting the version number

**Comment by WeatherGod** [comment]
Or perhaps do an intervention with the pyparsing devs?

**Comment by jenshnielsen** [comment]
Yes that would be ideal. Pyparsing is currently on sourceforge and in SVN I think. It would be ideal to move it to github and add Travis or similar to it

**Comment by WeatherGod** [comment]
Dear pyparsing,

Thank you for giving me the chance to talk to you today. There are so many
things that I want to say to you and share with you but it basically comes
down to this: I’m worried about your release procedures, and it’s time to
seek treatment.

Before you began doing poor releases, we were so close. There was nothing I
couldn’t tell you, and we always had a great time together. You were my
best friend in so many ways, but after months of poor releases, I barely
see that happy, fun project anymore. You seem sad, you’re struggling with
your commits and with the stress caused by not using TravisCI, and every
day I see the problems that Sourceforge is causing you pile up higher and
higher. I’m worried every day that you’re going to end up in the pypi
wheelhouse or abandoned or forked. I want to do everything I can to help
you get back to the Python community – of just being okay without
subversion and Sourceforge.
If you choose to move to GitHub (or GitLab, for that matter), I promise to
do what we can to get you the help you need to set up TravisCI and
coveralls. Hopefully, we can get you onto the road to making good releases
again!

Sincerely,
matplotlib

**Comment by jenshnielsen** [comment]
It seems like the issue has been fixed with pyparsing 2.1.7
